### PR TITLE
Do not evaluate relationship on included records in REST

### PR DESCRIPTION
**Pull request by skateman**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: app/serializers/compliance_serializer.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  def initialize(resource, options = {})
+    options[:params] ||= {}
+    options[:params][:root_resource] = resource.class
+
+    super(resource, options)
+  end
+
+  class_methods do
+    def relationships_hash(record, _, _, _, params = {})
+      return nil if params[:root_resource] != record.class
```
Comment (skateman) [inline app/serializers/compliance_serializer.rb:17]: I'm not entirely sure about the `nil` here, see the JIRA issue for more...

Comment (vkrizan) [reply app/serializers/compliance_serializer.rb:17]: Empty Hash/JSON Object could be more appropriate.

This makes `relationships` empty in any case, even on top level items.

Comment (skateman) [reply app/serializers/compliance_serializer.rb:10]: That has been adjusted :wink: gonna change the `nil` to `{}` then...

File: app/serializers/compliance_serializer.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+# JSON serializer parent class
+module ComplianceSerializer
```
Comment (akofink) [inline app/serializers/compliance_serializer.rb:4]: Nit: It's not a class :laughing: But you could make it one if you'd like! `ApplicationSerializer` would be the rails pattern.

File: app/serializers/compliance_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  extend ActiveSupport::Concern
+  include ::JSONAPI::Serializer
+
+  class_methods do
+    def relationships_hash(record, _, _, _, params = {})
+      return {} if params[:root_resource] != record.class
+
+      super
+    end
```
Comment (akofink) [inline app/serializers/compliance_serializer.rb:13]: Maybe we should add a comment about this being hacky and dependent on jsonapi_serializer's implementation. It seems like it'd be pretty stable, though, as long as the method name doesn't change :see_no_evil:

File: spec/integration/profiles_spec.rb [hunk 4]
```diff
@@ -255,7 +255,7 @@
                        type: { type: :string },
                        id: ref_schema('uuid'),
                        attributes: ref_schema('benchmark'),
-                       relationships: ref_schema('benchmark_relationships')
+                       relationships: {}
```
Comment (akofink) [inline spec/integration/profiles_spec.rb:258]: This is required to be a schema. An always-empty object would look like this:

```suggestion
                       relationships: { type: :object, properties: {} }
```

**Review by akofink — APPROVED** [review]
LGTM! Just a couple minor comments.

**Comment by akofink** [comment]
I forgot to check the openapi.json. If there are any included resource relationships there (I haven't checked), we should also update them to empty objects `{}`.

**Pull request merged**